### ISSUE TITLE
Regional hourly profiles for demand and supply are inconsistent with demand and supply volumes.  

### ISSUE THREAD

**Issue by AlexanderWirtz** [quintel/etsource#1933]
Our national dataset for `nl` is based on 2015 data. The regional datasets are based on 2016 energy data (volume), but use 2015 profiles. This means in principle the volumes and profiles are inconsistent. 

For smaller areas this is probably not so significant, but for larger ones, like Groningen + Drenthe, I expect this will prove problematic on closer inspection of the ETM's output. Martijn Douwes asked me about this the other day. Part of the attraction for him of using the ETM for the Groningen+Drenthe project is that the ETM outputs hourly profiles as well as demands. I am thinking about:

- Temperature profiles and heat demand
- solar PV profiles and solar PV power production
- wind profiles for onshore and offshore and wind power production
- ... did I forget something?

Since at some point we will have to update the national dataset anyway (let's hope we can find some budget for this), it seems like a good idea to research or create 2016 demand and supply profiles. 

According to @michieldenhaan the 2015 national profiles are now symlinked in the regional datasets. Is there some way to create a partial 2016 national dataset and symlink to the profiles there, so we can use them. This would allow us to use consistent profiles.  

Curious what you think @ChaelKruip, @michieldenhaan @jorisberkhout, @antw

**Comment by AlexanderWirtz** [comment]
Can this perhaps be done from the Stedin budget?

**Comment by michieldenhaan** [comment]
I don't think so, we only have a couple of days left. But maybe as part of the time-resolved update @ChaelKruip ?

**Comment by DorinevanderVlies** [comment]
Yes, it will be part of that project

**Comment by marliekeverweij** [comment]
This issue is fixed during the curves project! Now all regional datasets have 2016 profiles and the NL dataset uses 2015 profiles.

There is one important exception:
Because of incomplete data for wind and solar in 2015, the NL 2015 dataset will use 2016 profiles for solar PV and wind onshore and 2017 profile for wind offshore. This is also the case for regional datasets.

**Comment by AlexanderWirtz** [comment]
>Because of incomplete data for wind and solar in 2015, the NL 2015 dataset will use 2016 profiles for solar PV and wind onshore and 2017 profile for wind offshore. This is also the case for regional datasets.

@marliekeverweij I am not too happy with this fix, as this means wind data is internally inconsistent. I am sure there is a correlation with wind onshore and offshore.

I am also not so happy about the lack of consistency between solar data and wind data. I do not know about the correlation, but I assume it exists. Why not take everything from the same year? 

On top of that, there is no consistency between profile data and actual energy production in 2015.

In short: in my opinion, this is not an improvement to the ETM NL 2015 dataset. We kind of broke that. If the source of this data has so many gaps, why are we using it?

**Comment by AlexanderWirtz** [comment]
notifying @DorinevanderVlies @ChaelKruip @michieldenhaan

**Comment by DorinevanderVlies** [comment]
@AlexanderWirtz thanks for sharing your thoughts and concerns!

### Available data:
![image](https://user-images.githubusercontent.com/19907658/60095051-f7f7d100-974d-11e9-9d36-d4a2843d7b79.png)
In 2016 the offshore capacity increase significantly. So the measured output shows an upward trend (x3 in one year) which is unrealistic for 2030 and 2050. (see https://github.com/quintel/etmodel/issues/3058). 

### Options:
I think we have four options. Below I order them to my preference, I am not in favor or option 3 and 4. Please share your thoughts on this in comments or use the voting system. 

1. **Use 2017 curves as basis for all NL datasets** :heart:
For 2017 we have a complete set of data which we can use without further processing
Advantage: all curves are consistent, no futher processing required
Disadvantage: not year-specific
2. **Fix offshore wind 2016 and use 2016 data for NL2016 and use 2017 data for NL2015 and NL2017**
Advantage: consistent sets for 2016 and 2017 :tada:
Disadvantage: further processing required (0.5 man day)
Disadvantage: the rake task import curves will overwrite this manual adjustment. This can be confusing if people change other curves. We can solve this by making an exception in the rake task, but I am not sure how much work this is (@redekok ?). An other option is to put 2017 curves in the input and output folder of NL2015 on ETDataset and document this in the readme's.
3. **Current solution: NL2015: PV2015, Onshore 2015, Offshore 2017 NL2016: PV2016, Onshore2016, Offshore 2017** 👍
Advantage: If a curve is available we use it
Disadvantage: internally inconsistent
4. **Use the current curves for NL2015 and new curves for NL2016 and NL2017** :rocket:
Disadvantage: the current NL2015 curves are not year specific and not based on measured, but on derived data (combing a wind profile and power curve and for sun just irradiance data (which is year specific btw)

**Comment by DorinevanderVlies** [comment]
Including @marliekeverweij and @MartLubben

**Comment by ChaelKruip** [comment]
All options are basically the question of how you value data-completeness versus consistency with the energy balance. For me the latter is the most important because the ETM shows energy volumes as well as capacities and FLHs and these need to be internally consistent. Gaps in data can be dealt with. It might not be pretty but no-one will criticise you for trying to improve bad data. 

I lean towards option 4 with the following notes:

> Use the current curves for NL2015 and new curves for NL2016 and NL2017

Looking at these options this one seems to be the correct approach for 2016 and 2017. Am I correct to assume that we then still have to apply the same 'fix' for the 2016 offshore profile?

I'm not happy that this option would mean that we're stuck with the old, badly researched, profiles for 2015 but this could make it an extra priority to move to 2017 data for NL right?

**Comment by DorinevanderVlies** [comment]
> Looking at these options this one seems to be the correct approach for 2016 and 2017. Am I correct to assume that we then still have to apply the same 'fix' for the 2016 offshore profile?

Yes

> I'm not happy that this option would mean that we're stuck with the old, badly researched, profiles for 2015 but this could make it an extra priority to move to 2017 data for NL right?

Yes 😄

**Comment by DorinevanderVlies** [comment]
Closed with https://github.com/quintel/etsource/pull/2066
